WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentsAlertPresenterAdaptor.swift

```diff
@@ -25,6 +26,13 @@ final class CardPresentPaymentsAlertPresenterAdaptor: CardPresentPaymentAlertsPr
                     self?.paymentEventSubject.send(.idle)
                 })
             ))
+        case .paymentError(error: CardReaderServiceError.intentCreation(let underlyingError), _, let cancelPayment):
+            paymentEventSubject.send(.show(
+                eventDetails: .paymentIntentCreationError(error: underlyingError, cancelPayment: { [weak self] in
+                    cancelPayment()
+                    self?.paymentEventSubject.send(.idle)
+                })
+            ))
```

Comment:
Arguably this extra error isn't required; we can decode everything we need from the `CardReaderServiceError` further in to POS.

However, I quite like it because it gives something specific that we should handle differently in the client code.

Reply:
Yes. The reason it's handled separately is that I want to provide a different action to exit the payment flow - edit order. Payment Intent Creation error is unique because it happens before a payment is made by the user but was classified as a payment error. Separating it at this point made sense to me, in a similar way as it's done with capture payment error.